### PR TITLE
First version

### PR DESCRIPTION
**Pull request by oraluben**
# Matrix

Matrix is a config fuzzer, i.e. generate a set of configurations for a program that need to be fuzzed.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+GOBUILD=$(GO) build
+
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
+PACKAGE_LIST := go list ./... | grep -vE "matrix/test|pkg/ptrace|zz_generated|vendor"
```
Comment (zhouqiang-cl) [inline Makefile]: why pkg/ptrace

File: pkg/context/context.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
+package context
+
```
Comment (zhouqiang-cl) [inline pkg/context/context.go:15]: Add licence?

File: pkg/context/context.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import (
+	"chaos-mesh/matrix/pkg/node"
+	"chaos-mesh/matrix/pkg/node/data"
+	"chaos-mesh/matrix/pkg/random"
+	"chaos-mesh/matrix/pkg/serializer"
+	"fmt"
```
Comment (zhouqiang-cl) [inline pkg/context/context.go]: group it?

File: pkg/context/context.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	result.Configs = make(map[serializer.Config]interface{})
+
+	for _, config := range c.Configs {
+		result.Configs[config.Config] = simpleRecGen(config.Hollow)
+	}
+
+	return result
+}
+
+func simpleRecGen(hollow interface{}) interface{} {
```
Comment (zhouqiang-cl) [inline pkg/context/context.go:42]: what means hellow

File: pkg/context/scheduler.go [hunk 5]
```diff
@@ -0,0 +1 @@
+package context
```
Comment (zhouqiang-cl) [inline pkg/context/scheduler.go]: Did we need this file?

File: pkg/node/abstract.go [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package node
+
+import (
+	"chaos-mesh/matrix/pkg/serializer"
+)
+
+type AbstractConfig struct {
```
Comment (zhouqiang-cl) [inline pkg/node/abstract.go:20]: add comment?

File: pkg/node/abstract.go [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+	Tag string
+	serializer.Config
+	Hollow interface{}
```
Comment (zhouqiang-cl) [inline pkg/node/abstract.go:23]: Hollow?

**Comment by zhouqiang-cl** [comment]
please add a description

File: pkg/context/context.go (hunk 4 above)
Comment (you06) [reply pkg/context/context.go:42]: I think it's "trick or treat" which generates random variables.

**Review by mahjonp — APPROVED** [review]
LGTM

**Pull request merged**